alacritty_terminal/src/ansi.rs

```diff
@@ -970,129 +973,68 @@ where
 
                 handler.clear_line(mode);
             },
-            'S' => handler.scroll_up(Line(arg_or_default!(idx: 0, default: 1) as usize)),
-            'T' => handler.scroll_down(Line(arg_or_default!(idx: 0, default: 1) as usize)),
-            'L' => handler.insert_blank_lines(Line(arg_or_default!(idx: 0, default: 1) as usize)),
-            'l' => {
+            ('S', None) => handler.scroll_up(Line(arg_or_default!(idx: 0, default: 1) as usize)),
+            ('T', None) => handler.scroll_down(Line(arg_or_default!(idx: 0, default: 1) as usize)),
+            ('L', None) => handler.insert_blank_lines(Line(arg_or_default!(idx: 0, default: 1) as usize)),
+            ('l', intermediate) => {
+                let is_private_mode = match intermediate {
+                    Some(b'?') => true,
+                    None => false,
+                    _ => unhandled!(),
+                };
                 for arg in args {
-                    let mode = Mode::from_primitive(private, *arg);
+                    let mode = Mode::from_primitive(is_private_mode, *arg);
```

Comment:
TIL: https://github.com/rust-lang/rust/issues/62358

Ignore since this is a nightly only opt-in feature for now.

Reply:
Very interesting, didn't know about this one. Thanks for the info!